pkg/deploy/api/validation/validation.go

```diff
@@ -39,6 +39,39 @@ func ValidateDeploymentConfigSpec(spec deployapi.DeploymentConfigSpec) field.Err
 	if spec.Template == nil {
 		allErrs = append(allErrs, field.Required(specPath.Child("template"), ""))
 	} else {
+		// if we have both an ICT defined and an empty Template->PodSpec->Container->Image field, we are going
```

Comment:
OK, I've pushed the change that removes deepCopy and introduces a mutate/restore semantic.  I updated the test case that confirms the container name change does not extend beyond the validate call *slightly*, but certainly did not get expansive, given @smarterclayton said he'd be ok with a test that was the same.